node/metrics/createQuote.ts

```diff
@@ -0,0 +1,106 @@
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+type UserData = {
+  orgId: string
+  costId: string
+  roleId: string
+}
+
+type SessionData = {
+  namespaces: {
+    profile: {
+      id: { value: string }
+      email: { value: string }
+    }
+    account: {
+      accountName: { value: string }
+    }
+  }
+}
+
+type CreateQuoteMetricParam = {
+  sessionData: SessionData
+  sendToSalesRep: boolean
+  userData: UserData
+  quoteId: string
+  quoteReferenceName: string
+  creationDate: string
+}
+
+type CreateQuoteFieldsMetric = {
+  cost_center_id: string
+  cost_center_name: string
+  buy_org_id: string
+  buy_org_name: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+type CreateQuoteMetric = Metric & { fields: CreateQuoteFieldsMetric }
+
+const getCostCenterById = async (id: string, ctx: Context) => {
+  const cost = await ctx.clients.organizations.getCostCenterById(id)
+
+  return cost?.data?.getCostCenterById?.name
+}
+
+const getOrganizationById = async (id: string, ctx: Context) => {
+  const cost = await ctx.clients.organizations.getOrganizationById(id)
+
+  return cost?.data?.getOrganizationById?.name
+}
+
+const buildQuoteMetric = async (
+  metricsParam: CreateQuoteMetricParam,
+  ctx: Context
+): Promise<CreateQuoteMetric> => {
+  const { namespaces } = metricsParam.sessionData
+  const accountName = namespaces.account.accountName.value
+  const userEmail = namespaces.profile.email.value
+
+  const names = await Promise.all([
```

Comment:
Is it worth getting just the name with the getOrganization and getCostCenter functions? I don't know if it's really necessary, maybe we could use just the ID.

If it is necessary, I suggest decomposing the parameter instead of using the array position.
```
const [orgName, costCenterName] = await Promise.all([
```

Reply:
That's a good question. I don´t think its really necessary, but the names are data that are more "available" to users and that will be used to present data (even for us, it will be easier to make some analysis if we get the names at hand)